tapchannel/commitment.go

```diff
@@ -1187,45 +1187,46 @@ func collectOutputs(a *Allocation,
 	return outputs, nil
 }
 
-// CreateSecondLevelHtlcPackets creates the virtual packets for the second level
-// HTLC transaction.
-func CreateSecondLevelHtlcPackets(chanState *channeldb.OpenChannel,
-	commitTx *wire.MsgTx, htlcAmt btcutil.Amount,
-	keys lnwallet.CommitmentKeyRing, chainParams *address.ChainParams,
-	htlcOutputs []*cmsg.AssetOutput) ([]*tappsbt.VPacket, []*Allocation,
-	error) {
+// createSecondLevelHtlcAllocations creates the allocations for the second level
+// HTLCs. This will be used to generate the vPkts that corresponds to the second
+// level HTLC sweep.
+func createSecondLevelHtlcAllocations(chanType channeldb.ChannelType,
+	initiator bool, htlcOutputs []*cmsg.AssetOutput, htlcAmt btcutil.Amount,
+	commitCsvDelay uint32, keys lnwallet.CommitmentKeyRing,
+	outputIndex fn.Option[uint32],
+) ([]*Allocation, error) {
+
+	// TODO(roasbeef): thaw height not implemented for taproot chans rn
+	// (lease expiry)
```

Comment:
IIUC the only gap to lease expiration (from the PoV of `tapd`) is that sometimes the settled commitment output also has a CLTV. So that should just manifest as setting the CLTV in the sweep vPkt, then generating the correct script based on the bool of if it's leased enforced or not.